extensions/rates/functions/src/index.ts

```diff
@@ -0,0 +1,59 @@
+import * as functions from "firebase-functions";
+import ShipEngine, {  } from 'shipengine';
+
+import config from './config';
+import * as logs from './logs';
+import { Params } from "./types";
+
+interface InputPayload {
+  [key: string]: any
+}
+
+// Initialize ShipEngine client
+const shipEngine = new ShipEngine(config.shipEngineApiKey);
+
+logs.init();
+
+export const getRates = functions.handler.firestore.document.onWrite(
+  async (change) => {
+    logs.start();
+
+    const data = change.after.data() as InputPayload;
+    const params = data[config.inputKey] as Params;
```

Comment:
We *might* want to change how we handle the shipment details...I'm waffling between two approaches in the purchaseLabel extension, since I think we probably need to have it configurable enough to get the individual pieces of information for a shipment however they might design the data. We can make the assumption that they have it defined to match the shipment type structure, or we can break it down piece-by-piece with input keys in the config. If I come up with a reasonable solution in the purchaseLabel stuff, I'll back-port it here.